File: src/main/py/hello_word.py
from pyspark.sql import SparkSession
from pyspark.sql.types import Row

# Create the spark session
spark = SparkSession.builder.master("local").appName("count-words-python").getOrCreate()
text = spark.read.text("/path/to/hello.in")
words = text.rdd.flatMap(lambda line: line.value.split(" "))
large_words_df = words.filter(lambda w: len(w) > 2).map(lambda line: Row(line)).toDF()
large_words_df.write.text("/path/to/large-out-py")
spark.stop()

File: src/main/scala/com/gigahex/samples/HelloWord.scala
package com.gigahex.samples

import java.lang.System.Logger

import org.apache.commons.logging.LogFactory
import org.apache.spark.sql.SparkSession

object HelloWord {

  val logger = LogFactory.getLog(getClass)

  def main(args: Array[String]): Unit = {
    /**
     * Setup the spark session
     */
    val spark = SparkSession.builder()
      .appName("count-words-scala")
      .getOrCreate()
    import spark.implicits._

    val text = spark.read.textFile("/Users/gigahex/hello.in")
    val words = text.flatMap(x => x.split(" "))
    val largeWords = words.filter(w => w.length > 2)
    val outputPath = s"/Users/gigahex/large-words-scala/${System.currentTimeMillis()}"
    largeWords.write.text(outputPath)
    spark.read.textFile(s"${outputPath}/*").show(30)

    spark.stop()
  }

}
